### PR TITLE
1303 refactor endpoints

### PR DESCRIPTION
**Pull request by BCerki**
https://github.com/orgs/bcgov/projects/122/views/2?pane=issue&itemId=59073618
happo diff is flake

### CODE REVIEW

File: bc_obps/service/operator_service.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        elif legal_name:
+            try:
+                operators = OperatorDataAccessService.get_operators_by_legal_name(legal_name)
+                return [OperatorSearchOut.model_validate(operator) for operator in operators]
+            except:
+                raise Exception("No matching operator found. Retry or add operator.")
+
+    @classmethod
+    def archive_parent_operators(
+        cls, existing_parent_operators: List[Operator], updated_parent_operators: List[Operator], user_guid: UUID
```
Comment (Sepehr-Sobhani) [inline bc_obps/service/operator_service.py:43]: The type here should be `List[ParentOperator]`.

**Review by Sepehr-Sobhani — CHANGES_REQUESTED** [review]
Looks great! 🤩  Just a quick change is needed.

File: bc_obps/registration/api/operators.py [hunk 2]
```diff
@@ -26,37 +22,17 @@
 def get_operators_by_cra_number_or_legal_name(
```
[the remark this answers is not in the document]
Comment (BCerki) [reply bc_obps/registration/api/operators.py:23]: Thanks for catching this!

**Review by Sepehr-Sobhani — APPROVED** [review]
🥇

**Pull request merged**